ProcessLib/HeatConduction/HeatConductionFEM-impl.h

```diff
@@ -0,0 +1,155 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2017, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ *  \file   HeatConductionFEM-impl.h
+ *  Created on January 17, 2017, 3:41 PM
+ */
+
+#pragma once
+
+#include "HeatConductionFEM.h"
+#include "NumLib/Function/Interpolation.h"
+
+namespace ProcessLib
+{
+namespace HeatConduction
+{
+template <typename ShapeFunction, typename IntegrationMethod,
+          unsigned GlobalDim>
+template <typename LiquidFlowVelocityCalculator>
+void LocalAssemblerData<ShapeFunction, IntegrationMethod, GlobalDim>::
+    assembleHeatTransportLiquidFlow(
+        double const t, int const material_id, SpatialPosition& pos,
+        int const gravitational_axis_id,
+        double const gravitational_acceleration, Eigen::MatrixXd const& perm,
+        ProcessLib::LiquidFlow::LiquidFlowMaterialProperties const&
+            liquid_flow_prop,
+        std::vector<double> const& local_x, std::vector<double> const& local_p,
+        std::vector<double>& local_M_data, std::vector<double>& local_K_data)
+{
+    auto const local_matrix_size = local_x.size();
+    // This assertion is valid only if all nodal d.o.f. use the same shape
+    // matrices.
+    assert(local_matrix_size == ShapeFunction::NPOINTS * NUM_NODAL_DOF);
+
+    auto local_M = MathLib::createZeroedMatrix<NodalMatrixType>(
+        local_M_data, local_matrix_size, local_matrix_size);
+    auto local_K = MathLib::createZeroedMatrix<NodalMatrixType>(
+        local_K_data, local_matrix_size, local_matrix_size);
+    const auto local_p_vec =
+        MathLib::toVector<NodalVectorType>(local_p, local_matrix_size);
+
+    unsigned const n_integration_points =
+        _integration_method.getNumberOfPoints();
+
+    const double porosity_variable = 0.;
+    for (unsigned ip = 0; ip < n_integration_points; ip++)
+    {
+        pos.setIntegrationPoint(ip);
+        auto const& sm = _shape_matrices[ip];
+        auto const& wp = _integration_method.getWeightedPoint(ip);
+        double p = 0.;
+        NumLib::shapeFunctionInterpolate(local_p, sm.N, p);
+        double T = 0.;
+        NumLib::shapeFunctionInterpolate(local_x, sm.N, T);
+
+        // Material parameters of solid phase
+        auto const k_s = _process_data.thermal_conductivity(t, pos)[0];
+        auto const cp_s = _process_data.heat_capacity(t, pos)[0];
+        auto const rho_s = _process_data.density(t, pos)[0];
+
+        // Material parameters of fluid phase
+        double const cp_f = liquid_flow_prop.getHeatCapacity(p, T);
+        double const k_f = liquid_flow_prop.getThermalConductivity(p, T);
+        double const rho_f = liquid_flow_prop.getLiquidDensity(p, T);
+
+        // Material parameter of porosity
+        double const poro =
```

Comment:
could you please change to long variable names, where it is feasible?
porosity, permeability. (or use short names (n, k), but then with comments...)